src/java.base/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -87,7 +90,28 @@ public static MemorySegment alignUp(MemorySegment ms, long alignment) {
         return ms.asSlice(alignUp(offset, alignment) - offset);
     }
 
-    public static VarHandle makeSegmentViewVarHandle(ValueLayout layout) {
+    /**
+     * This method returns a <em>raw var handle</em>, that is, a var handle that does not perform any size
+     * or alignment checks. Such checks are added (using adaptation) by {@link LayoutPath#dereferenceHandle()}.
+     * <p>
+     * We provide two level of caching of the generated var handles. First, the var handle associated
+     * to a {@link ValueLayout#varHandle()} call is cached inside a stable field of the value layout implementation.
+     * This optimizes common code idioms like {@code JAVA_INT.varHandle().getInt(...)}. A second layer of caching
+     * is then provided by this method: after all, var handles constructed by {@link MemoryLayout#varHandle(PathElement...)}
+     * will be obtained by adapting some raw var handle generated by this method.
+     *
+     * @param layout the value layout for which a raw memory segment var handle is to be created.
+     * @return a raw memory segment var handle.
+     */
+    public static VarHandle makeRawSegmentViewVarHandle(ValueLayout layout) {
```

Comment:
Since I was here, I took the opportunity to rename this to include `raw`, to denote that the var handle returned by this method should not be exposed by APIs as is.